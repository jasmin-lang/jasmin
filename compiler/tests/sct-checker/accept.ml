open Jasmin
open Common
open Sct_checker_forward

let path = "success"

let load_and_check name =
  Format.printf "File %s:@." name;
  let p = load_file (Filename.concat path name) in
  match ty_prog Arch.is_ct_sopn p [] with
  | exception Utils.HiError e ->
      Format.eprintf "%a@." Utils.pp_hierror e;
      exit 1
  | r -> List.iter (Format.printf "%a@." pp_funty) r

let () =
  let cases = Sys.readdir path in
  Array.sort String.compare cases;
  Array.iter load_and_check cases
