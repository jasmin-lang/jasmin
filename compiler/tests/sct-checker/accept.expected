File annot.jazz:
modmsf f : #transient ->
#secret
output corruption: #public
 constraints:


modmsf id : #poly = { n = k, s = k} ->
#poly = { n = k, s = k}
output corruption: #public
 constraints:
k <= k, 

nomodmsf id1 : #poly = { n = k, s = k} ->
#poly = { n = k, s = k}
output corruption: #public
 constraints:


nomodmsf id2 : #[ptr = { n = p, s = p}, val = { n = v, s = v}] ->
#[ptr = { n = p, s = p}, val = { n = v, s = v}]
output corruption: #public
 constraints:


nomodmsf id3 : #public ->
#public
output corruption: #public
 constraints:


nomodmsf id4 : #transient ->
#transient
output corruption: #public
 constraints:


nomodmsf id5 : #secret ->
#secret
output corruption: #public
 constraints:


nomodmsf unused_msf : #msf ->

output corruption: #public
 constraints:


File arrays.jazz:
modmsf transient_read : #[ptr = transient, val = { n = d, s = secret}] *
                        #transient ->
#poly = { n = d, s = secret}
output corruption: #public
 constraints:


modmsf safe_access_no_array : #public * #poly = { n = d, s = d} ->
#poly = { n = d, s = d}
output corruption: #public
 constraints:


modmsf safe_access : #public * #poly = { n = d, s = secret} ->
#poly = { n = d, s = secret}
output corruption: #public
 constraints:


modmsf safe_direct_access : #public * #poly = { n = d, s = secret} ->
#poly = { n = d, s = secret}
output corruption: #public
 constraints:


File basic.jazz:
modmsf nothing : #transient ->
#public
output corruption: #public
 constraints:


nomodmsf secret : #secret ->
#secret
output corruption: #public
 constraints:


modmsf branch : #transient * #secret * #secret ->
#secret
output corruption: #public
 constraints:


nomodmsf branchless : #transient * #secret * #secret ->
#secret
output corruption: #public
 constraints:


modmsf forloop : #transient ->
#public
output corruption: #public
 constraints:


modmsf whileloop : #transient ->
#public
output corruption: #public
 constraints:


modmsf archetype : #transient ->
#transient
output corruption: #public
 constraints:


modmsf load : #transient * #transient ->
#public
output corruption: #public
 constraints:


modmsf store : #transient * #transient * #transient * #secret ->

output corruption: #transient
 constraints:


nomodmsf xof_init : #public * #[ptr = public, val = transient] *
                    #[ptr = public, val = secret] ->
#[ptr = public, val = secret]
output corruption: #transient
 constraints:


File bug_852.jazz:
nomodmsf reset : #public ->

output corruption: #public
 constraints:


modmsf main : #transient ->

output corruption: #public
 constraints:


File bug_887.jazz:
modmsf test_msf : #secret * #transient ->

output corruption: #public
 constraints:


modmsf test_venv : #secret ->
#public
output corruption: #public
 constraints:


File corruption.jazz:
nomodmsf corrupts_memory : #public * #secret *
                           #[ptr = public, val = secret] *
                           #[ptr = public, val = public] ->
#[ptr = public, val = secret] * #[ptr = public, val = transient]
output corruption: #transient
 constraints:


nomodmsf does_not_corrupt_memory : #secret * #[ptr = public, val = secret] *
                                   #[ptr = public, val = public] ->
#[ptr = public, val = secret] * #[ptr = public, val = public]
output corruption: #public
 constraints:


modmsf does_corrupt_memory : #secret * #[ptr = public, val = secret] *
                             #[ptr = public, val = public] ->
#[ptr = public, val = secret] * #[ptr = public, val = transient]
output corruption: #transient
 constraints:


nomodmsf does_not_corrupt_registers : #public * #secret *
                                      #[ptr = public, val = secret] ->
#[ptr = public, val = secret] * #public
output corruption: #transient
 constraints:


File for_loop.jazz:
modmsf for_inference_bug1 : #msf ->
#msf
output corruption: #public
 constraints:


modmsf for_inference_bug2 : #transient ->
#public
output corruption: #public
 constraints:


File local-stack-array.jazz:
modmsf main :  ->
#public
output corruption: #public
 constraints:


File movmsf.jazz:
modmsf reset_msf :  ->
#msf
output corruption: #public
 constraints:


modmsf main : #transient ->

output corruption: #public
 constraints:


File paper.jazz:
modmsf fig3a : #[ptr = transient, val = transient] *
               #[ptr = transient, val = secret] * #transient ->
#[ptr = public, val = secret]
output corruption: #public
 constraints:


modmsf fig3b : #[ptr = transient, val = transient] *
               #[ptr = transient, val = { n = w, s = secret}] *
               #[ptr = transient, val = secret] * #transient * #secret ->
#[ptr = public, val = { n = w, s = secret}] * #[ptr = public, val = secret]
output corruption: #transient
 constraints:


modmsf fig4a : #[ptr = public, val = secret] * #[ptr = public, val = secret] ->
#[ptr = public, val = secret]
output corruption: #transient
 constraints:


modmsf fig4b : #[ptr = transient, val = secret] *
               #[ptr = transient, val = secret] ->
#[ptr = public, val = secret]
output corruption: #transient
 constraints:


modmsf fig4c : #[ptr = transient, val = secret] *
               #[ptr = transient, val = secret] ->
#[ptr = public, val = secret]
output corruption: #transient
 constraints:


modmsf fig5a : #[ptr = public, val = { n = d, s = secret}] ->
#poly = { n = d, s = secret}
output corruption: #public
 constraints:


modmsf fig5b : #[ptr = transient, val = { n = d, s = d}] ->
#poly = { n = d, s = d}
output corruption: #public
 constraints:
d <= d, 

modmsf fig5c : #[ptr = transient, val = { n = d, s = d}] ->
#poly = { n = d, s = d}
output corruption: #public
 constraints:
d <= d, 

modmsf fig6a : #[ptr = transient, val = secret] *
               #[ptr = transient, val = transient] * #transient * #transient ->
#public * #[ptr = public, val = secret]
output corruption: #public
 constraints:


modmsf fig6b : #[ptr = transient, val = secret] *
               #[ptr = transient, val = transient] * #transient * #secret ->
#public * #[ptr = public, val = secret]
output corruption: #public
 constraints:


File spill.jazz:
nomodmsf spill1 : #public * #secret ->
#public
output corruption: #transient
 constraints:


modmsf spill2 : #public * #secret ->
#public
output corruption: #transient
 constraints:


File while.jazz:
modmsf while_first_branch : #public * #secret ->
#public
output corruption: #public
 constraints:


